src/k8s/pkg/client/dqlite/remove_test.go

```diff
@@ -48,17 +48,23 @@ func TestRemoveNodeByAddress(t *testing.T) {
 			g.Expect(err).To(BeNil())
 			g.Expect(members).To(HaveLen(2))
 
-			memberToRemove := members[0].Address
+			memberToRemove := members[0]
+			remainingNode := members[1]
 			if members[0].Role != dqlite.Voter {
-				memberToRemove = members[1].Address
+				memberToRemove = members[1]
+				remainingNode = members[0]
 			}
+			g.Expect(memberToRemove.Role).To(Equal(dqlite.Voter))
+			g.Expect(remainingNode.Role).To(Equal(dqlite.Spare))
 
-			// Removing the last Voter should fail
-			g.Expect(client.RemoveNodeByAddress(ctx, memberToRemove)).ToNot(BeNil())
+			// Removing the last voter should succeed and leadership should be transfered.
+			g.Expect(client.RemoveNodeByAddress(ctx, memberToRemove.Address)).To(Succeed())
 
 			members, err = client.ListMembers(ctx)
 			g.Expect(err).To(BeNil())
-			g.Expect(members).To(HaveLen(2))
+			g.Expect(members).To(HaveLen(1))
+			g.Expect(members[0].Role == dqlite.Voter)
+			g.Expect(members[0].Address).ToNot(Equal(memberToRemove.Address))
```

Comment:
Do we have a test for edge cases like `remove the last node in the cluster`?

Reply:
Added a test that this fails.